### ISSUE TITLE
Cache on iOS

### ISSUE THREAD

**Issue by mariusmuntean** [luberda-molinet/FFImageLoading#239]
Hi!
I noticed that on iOS the disk cache(FImageLoading/source/FFImageLoading.Shared/Cache/SimpleDiskCache.cs) is created in the tmp/ subdirectory of the App.
I think this is not Ideal since, as per [Apple's documentation](https://developer.apple.com/library/ios/documentation/FileManagement/Conceptual/FileSystemProgrammingGuide/FileSystemOverview/FileSystemOverview.html), the content of that folder might not survive app restarts.

They recommend putting such data either in the Documents and the Library/Caches subdirectories.

We're now using a custom disk cache under Documents, but this means that we need to integrate a routine to migrate the cache from its old location to the new one so as not to degrade the user experience. Maybe you could implement something like moving the cache.

**Comment by daniel-luberda** [comment]
@mariusmuntean Temp is fine. It's only emptied under some circumstances as low-disk space and it's designed for use-cases like this. You can use your own path if you want to:

```C#
var config = new FFImageLoading.Config.Configuration()
{
	DiskCache = new Cache.SimpleDiskCache("[YOUR_PATH_HERE]")
};
ImageService.Instance.Initialize(config);
```

**Comment by daniel-luberda** [comment]
Currently we use `Path.GetTempPath()` method.

**Comment by molinch** [comment]
Dunno if we should really do anything since iOS will usually only reclaim /tmp space when the device is low on disk.

I am not against changing the path but if we do so we should be extremely careful about iCloud backup. Cache files should never ever be sent to iCloud otherwise Apple will reject the app.

**Comment by molinch** [comment]
@daniel-luberda Is right we can use /Library/Caches

**Apple documentation**
Library/
This is the top-level directory for any files that are not user data files. You typically put files in one of several standard subdirectories. iOS apps commonly use the Application Support and Caches subdirectories; however, you can create custom subdirectories.
Use the Library subdirectories for any files you don’t want exposed to the user. Your app should not use these directories for user data files.
**The contents of the Library directory (with the exception of the Caches subdirectory) are backed up by iTunes.**
For additional information about the Library directory and its commonly used subdirectories, see The Library Directory Stores App-Specific Files.
https://developer.apple.com/library/ios/documentation/FileManagement/Conceptual/FileSystemProgrammingGuide/FileSystemOverview/FileSystemOverview.html

**Comment by molinch** [comment]
@mariusmuntean If you want to provide a PR for it we are happy to merge it.

**Comment by daniel-luberda** [comment]
Should be fixed in `2.1.9-pre-168-core-r`. Please reopen it if doesn't work for you.
